presto-docs/src/main/sphinx/language/types.rst

```diff
@@ -217,8 +217,18 @@ See also :doc:`/functions/datetime` and :doc:`/language/timestamp`
 ``TIMESTAMP WITH TIME ZONE``
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-    Instant in time that includes the date and time of day with a time zone.
-    Values of this type are rendered using the time zone from the value.
+    ``TIMESTAMP WITH TIME ZONE`` is an alias for ``TIMESTAMP(3) WITH TIME ZONE``
+    (millisecond precision).
+
+``TIMESTAMP(P) WITH TIME ZONE``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    Instant in time that includes the date and time of day with ``P`` digits of
+    precision for the fraction of seconds and with a time zone. Values of this
+    type are rendered using the time zone from the user session.
+
+    Other behavior and usage is identical to :ref:`TIMESTAMP
+    <timestamp-data-type>`.
```

Comment:
This is not quite true. TIMESTAMP and TIMESTAMP WITH TIMEZONE are different types with different semantics. In standard SQL, TIMESTAMP is more like DATE, with additional precision that includes hour/minute/second/fraction. TIMESTAMP W/ TZ represents an instant with an attached time zone.

Reply:
So then we should probably expand hat section with relevant info and add more examples. Do you know what we should add specifically

Reply:
It's complicated due to the fact that we support a legacy form of timestamp that doesn't follow standard SQL semantics. Maybe we need to call that out explicitly and explain how legacy timestamps work in Presto. @electrum, thoughts?

Reply:
We removed the sentence altogether for now.